### ISSUE TITLE
Think about surface datasets for very high resolution cases...

### ISSUE THREAD

**Issue by ekluzek** [ESCOMP/CTSM#1292]
@patc is running into some trouble for very high resolution cases (global 3 km). One thing we need to think about for these type of cases is that the surface dataset (and landuse.timeseries file) will hold data that is interpolated from much lower resolution. So the files will be very large and unwieldy, but the data it represents is really smooth as it comes from lower resolution datasets. For these very high resolution cases it probably makes more sense to have surface datasets at some resolution that reflects the data it's coming from, and then have the model do some sort of interpolation or fill at initialization to the very high resolution grid that's actually being run.

There are lists of datasets that we do this for (the streams datasets) where interpolation is done on the fly. The thing here would be to also do it for surface datasets. This also might require moving some of the mksurfdata_map code into CTSM itself. Some fields on the surface dataset could be converted to stream files, but some, the fields that determine the sub-grid weights, have to handled more carefully as they need to sum to exactly 100% in each gridcell.

We don't have resources to dedicate to this issue yet. We would need help from projects that are doing high resolution work to get this to happen.

**Comment by ekluzek** [comment]
One new thought @mvertens had on this is that we use single-precision for our files rather than double. We originally had a bunch of trouble trying to do previously in CLM3 and CLM4. I think these problems all have to do with the "PCT" variables that have to sum to 100 and will be double precision inside of CTSM. So we possibly could make all but the PCT variables single-precision and likely get it to work that way. The problem we found with single precision is that we could make the single-precision files, but they would fail when we ran in CLM (at least for some resolutions). There are some variables that are written out as double that are also actually integers, so they could be changed to shorts to save space as well.
